security/src/main/java/pm/gnosis/svalinn/security/impls/AesEncryptionManager.kt

```diff
@@ -82,26 +76,24 @@ class AesEncryptionManager(
         else
             SCrypt.generate(password, Sha3Utils.sha3(password), passwordIterations, SCRYPT_BLOCK_SIZE, SCRYPT_PARALLELIZATION, SCRYPT_KEY_LENGTH)
 
-    override fun setupPassword(newPassword: ByteArray, oldPassword: ByteArray?): Single<Boolean> {
-        return Single.fromCallable {
-            synchronized(keyLock) {
-                val checksum = preferencesManager.prefs.getString(PREF_KEY_PASSWORD_CHECKSUM, null)
-                var previousKey: ByteArray? = null
-                if (checksum != null) {
-                    previousKey = buildPasswordKeyIfValid(oldPassword, checksum) ?: return@fromCallable false
-                }
-                val passwordKey = deriveKeyFromPassword(newPassword)
-                key = previousKey?.let {
-                    decryptAppKey(it)
-                } ?: generateKey()
-                key?.let {
-                    preferencesManager.prefs.edit { putString(PREF_KEY_PASSWORD_ENCRYPTED_APP_KEY, encrypt(passwordKey, it).toString()) }
-                    preferencesManager.prefs.edit { putString(PREF_KEY_PASSWORD_CHECKSUM, generateCryptedChecksum(passwordKey)) }
-                }
-
-                key != null
+    override fun setupPassword(newPassword: ByteArray, oldPassword: ByteArray?): Boolean {
```

Comment:
A bit concerned that this is no longer called in a different thread. Couldn't we mark it as `suspend` to enforce this being from a coroutine context?

Reply:
Imo that is up to the developer that is using the lib ... not sure if I would enforce this if not 100% necessary

Reply:
I was thinking though, the dev could still execute this in the main thread (if for some reason necessary) but then it would also have to be explicit from which context you launch the coroutine. This is just a nudge, kind of like saying "maybe you should do this in a different thread". Either way, it is not super important.